ambry-router/src/main/java/com.github.ambry.router/GetOperation.java

```diff
@@ -172,9 +153,9 @@ void processServerError(ServerErrorCode errorCode) {
    * @param exception the {@link RouterException} to possibly set.
    */
   void setOperationException(RouterException exception) {
-    if (operationException == null || exception.getErrorCode() == RouterErrorCode.BlobDeleted
+    if (operationException.get() == null || exception.getErrorCode() == RouterErrorCode.BlobDeleted
```

Comment:
Why wouldn't we? That is the unambiguous case - when the first exception comes in we simply set it no matter what.

Note that setting this exception does not necessarily mean the operation has completed. This method is called every time an exception is received for a request, we may continue to do more requests and this method can get called over and over. Eventually the most relevant exception gets set. If `Blob_Deleted` or `Blob_Expired` is received, the operation completes immediately. If a successful response comes in, `operationException` is cleared before the operation is completed.